step-04/index.js

```diff
@@ -29,15 +29,15 @@ io.sockets.on('connection', function(socket) {
   socket.on('create or join', function(room) {
     log('Received request to create or join room ' + room);
 
-    var numClients = io.sockets.sockets.length;
+    var clientsInRoom = io.sockets.adapter.rooms[room];
```

Comment:
Actually — I was wrong. Works like this for me...

One client:

`{ jUeCl8VQyK1L0URZAAAB: { jUeCl8VQyK1L0URZAAAB: true } }`

Two clients:

`{ jUeCl8VQyK1L0URZAAAB: { jUeCl8VQyK1L0URZAAAB: true },
  foo: { jUeCl8VQyK1L0URZAAAB: true },
  HG1inGcNxv1GaXMCAAAC: { HG1inGcNxv1GaXMCAAAC: true } }`